taxclock/scrapers/base.py

```diff
@@ -63,23 +74,72 @@ def aws_store(self, data):
             data = self.sort_data_by_date(data)
             try:
                 self.s3.put_object(
+                    # TODO: Pull bucket from settings
                     Bucket=env.AWS_S3_BUCKET,
                     ACL='public-read',
                     Key='data/news.json',
                     Body=json.dumps(data[:7]))
             except Exception as err:
                 self.local_store(data[:7])
                 log.info(str(err))
+                self.error_details['error'] = str(err)
+                self.error_details['site_url'] =\
+                    "S3 is not available for all urls"
+                self.error_details['severity'] = env.ERROR_SEVERITY[1]
+                self.send_error_log(self.error_details)
         else:
             log.error('No data to save')
 
     def sort_data_by_date(self, data):
         '''Sorts data by date.
         Usage::
             Pass the data to sort.
-        :rtype: outputs the sorted data.
+        :rtype: outputs a list of sorted data.
         '''
         if not data:
             log.info('No data was found for sorting.')
         return sorted(data, key=lambda k: parse(k['date_published']),
                       reverse=True)
+
+    def format_error(self, error):
+        '''Formats the error message
+        Usage::
+            pass the error
+        :rtype: returns the message well formatted.
+        '''
+        message = json.dumps({
+            'origin': 'Taxclock Scraper',
+            'origin_url': error['site_url'],
+            'error_details': [
+                {
+                    'error': error['error'],
+                    'severity': error['severity'],
+                    'labels': {
+                        'Product': 'TaxClock Kenya',
+                        'Team': 'Code for Kenya Team',
+                    },
+                    'type': 'Web Scraping Application',
+                }
+            ]
+        }, indent=2)
+        return message
+
+    def send_error_log(self, error):
+        '''Sends message to slack
+        Usage::
+           pass the error message.
+        :rtype: outputs the error to slack.
+        '''
+        slack_handler = SlackerLogHandler(env.SLACK_CHANNEL_TOKEN,
+                                          env.SLACK_CHANNEL_NAME,
+                                          stack_trace=True)
+        # Create logger
+        logger = logging.getLogger('debug_application')
+        logger.addHandler(slack_handler)
+        formatter = NoStacktraceFormatter(
+            "{'log_date':%(asctime)s, 'message': %(message)s}")
+        slack_handler.setFormatter(formatter)
+        # Define the minimum level of log messages you want to send to Slack
+        slack_handler.setLevel(logging.DEBUG)
+        # Test logging
```

Comment:
Let's have this in the `scraper.py` file so that we're not having to set each time we are logging an error.

That way when logging we only do `log.error(error)`.

Reply:
@DavidLemayian  Looking into ways to achieve this.